test/utils/utils.go

```diff
@@ -263,7 +286,173 @@ func VerifyConfigMapContents(c client.Client, name, namespace, key, expect strin
 	return data == expect
 }
 
+// StartPortForward initiates a port forwarding connection to a pod on the
+// localhost interface. It returns a closer function that should be invoked to
+// stop the proxy server.
+// The function blocks until the port forwarding proxy server is ready to
+// receive connections or the context is canceled.
+func StartPortForward(ctx context.Context, port intstr.IntOrString, scheme, name, ns string) (func(), error) {
```

Comment:
So with this, I think we are doing a lot of work. I wonder if there is some test framework already which provides things like this?

Also, maybe we can simplify quite a bit here, and just do service portforward by executing kubectl command? Should work equivalentely